internal/aggregated-co/writer.go

```diff
@@ -0,0 +1,66 @@
+package aggregated_co
+
+import (
+	"context"
+	"reflect"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// From https://github.com/operator-framework/operator-lifecycle-manager/pkg/lib/operatorstatus/writer.go
+
+// NewWriter returns a new instance of Writer.
+func NewWriter(client configv1client.ConfigV1Interface) *Writer {
+	return &Writer{
+		client: client,
+	}
+}
+
+// Writer encapsulates logic for cluster operator object API. It is used to
+// update ClusterOperator resource.
+type Writer struct {
+	client configv1client.ConfigV1Interface
```

Comment:
nit: We could refactor this to use c-r's dynamic client, and add the configv1 scheme to a central scheme?

Reply:
This about this more: I think this is less of a nit and more of a potential blocker here. We're introducing a dependency that we don't necessarily need, and inflating the changeset of this PR given we're committing the vendor tree in this repository.

Reply:
I'm not sure I see the benefit of switching to the dynamic client here -- having the typed client in tree will be helpful, especially as we will have to manage the POM CO at some point too. We do vendor a bit, but just the typed client we need essentially.